lib/src/layouts/yaru_compact_layout.dart

```diff
@@ -35,33 +40,62 @@ class YaruCompactLayout extends StatefulWidget {
   final IndexedWidgetBuilder pageBuilder;
 
   /// The index of the initial page to show.
-  final int initialIndex;
+  final int? initialIndex;
 
   /// Called when the user selects a page.
   final ValueChanged<int>? onSelected;
 
+  /// An optional controller that can be used to navigate to a specific index.
+  final YaruPageController? controller;
+
   @override
   State<YaruCompactLayout> createState() => _YaruCompactLayoutState();
 }
 
 class _YaruCompactLayoutState extends State<YaruCompactLayout> {
   late int _index;
```

Comment:
do we need a separate `_index` anymore because a controller always exists so maybe we could pass `_pageController.index` to the layout? the less we have these state variables the better because they all need to be synced in `didUpdateWidget()` :)

Reply:
I kept it to deal with a negative index (which is only really needed in `YaruPortraitLayout`), but it's simpler to deal with this in the appropriate places than taking care of another state variable :+1: